src/experimental/ProbabilisticGraphicalModels/bayesian_network.jl

```diff
@@ -161,3 +164,39 @@ function add_edge!(bn::BayesianNetwork{V,T}, from::V, to::V)::Bool where {T,V}
     to_id = bn.names_to_ids[to]
     return Graphs.add_edge!(bn.graph, from_id, to_id)
 end
+
+function evaluate(bn::BayesianNetwork)
+    logp = 0.0
+    evaluation_env = bn.evaluation_env
+
+    for (i, varname) in enumerate(bn.names)
+        is_stochastic = bn.is_stochastic[i]
+        if is_stochastic
+            dist_fn = bn.distributions[i](evaluation_env, bn.loop_vars)
+
+            value = getproperty(evaluation_env, Symbol(varname))
+            bijector = Bijectors.bijector(dist_fn)
+            value_transformed = Bijectors.transform(bijector, value)
+
+            logpdf_val = Distributions.logpdf(dist_fn, value)
+            logjac = Bijectors.logabsdetjac(Bijectors.inverse(bijector), value_transformed)
+            logp += logpdf_val + logjac
+
+        else
+            fn = bn.deterministic_functions[i](evaluation_env, bn.loop_vars)
+            parent_vals = parent_values(bn, evaluation_env, i)
```

Comment:
this is no longer needed, because now we don't store values for each variables, but managing all the values in `evaluation_env`